addons/website_blog/controllers/main.py

```diff
@@ -169,8 +169,10 @@ def blog(self, blog=None, tag=None, page=1, **opt):
             values['main_object'] = blog
             values['edit_in_backend'] = True
             values['blog_url'] = QueryURL('', ['blog', 'tag'], blog=blog, tag=tag, date_begin=date_begin, date_end=date_end)
-        else:
+        elif tag:
             values['blog_url'] = QueryURL('/blog', ['tag'], date_begin=date_begin, date_end=date_end)
+        else:
+            values['blog_url'] = QueryURL('', ['blog'], date_begin=date_begin, date_end=date_end)
```

Comment:
I still don't think this code makes sense @Guillaume-gdi ... I am not an expert of QueryURL, never used it but as you had to understand it, you now should be an expert:
- The first one `QueryURL('', ['blog', 'tag'], blog=blog, tag=tag, date_begin=date_begin, date_end=date_end)` is called when there is a `blog`, and apparently works wheter or not there is a `tag` (since we don't check for its existence)
- The second one you add `QueryURL('/blog', ['tag'], date_begin=date_begin, date_end=date_end)` is called when there is no blog but you check there is a tag... why would it be needed since apparently the first one is able to understand to not add `/tag/my-tag` when there is no tag? (= why do we need a special case for "no blog but a tag" but not for "blog but no tag" ?)

Anyways, this needs more explanations and this is urgent, please test everything before asking for a review. Here I tested half a minute and found something that changed because of this PR:
- Install blog and go to `/blog`
- Click on a tag, like "hotels"

Before this PR: 
- you land on `/blog/tag/hotels-1`

After this PR:
- you land on `/blog?tag=hotels-1`

This works of course but this is still a change in behavior not acceptable for stable unless there is no other way. And here I really think there is another way: the code should really be about one "blog_url" method instead of 2. Not 3 instead of 2 :grimacing: